editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -3827,6 +3831,8 @@ void CanvasItemEditor::_draw_viewport() {
 	transform.columns[2] = -view_offset * zoom;
 	EditorNode::get_singleton()->get_scene_root()->set_global_canvas_transform(transform);
 
+	_update_lock_and_group_button();
```

Comment:
You could instead update when the button was pressed.

Reply:
But that means add update to four different places (lock, unlock, group, ungroup), and deal with UndoRedo carefully, IMHO it doesn't seem very neat

Reply:
And the alternative is updating the buttons every time the viewport redraws, for no reason, because most actions don't affect selection.

There are signals `item_lock_status_changed` and `item_group_status_changed`. You can just connect them to update buttons.

Reply:
Done